##### Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day.
##### You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. 
##### Assume prior research suggests the population standard deviation is 2.3 minutes.
##### Conduct a test with a level of significance of alpha = 0.05.

In [1]:
import scipy.stats as stats
import numpy as np


In [2]:
population_mean=11  # population mean under null hypothesis
sample_size=1000    
sample_mean=11.5
population_std=2.3  # standard deviation of population
alpha=0.05          # significance level

In [3]:
# calculate z test statistics and critical value
z_score=(sample_mean-population_mean)/(population_std/np.sqrt(sample_size))
critical_value=stats.norm.ppf(1-alpha)
# print the z score and critical value
print("Z_score : ",z_score)
print("Critical_value :",critical_value)

Z_score :  6.874516652539955
Critical_value : 1.6448536269514722


In [4]:
# compare the z score with the critical value and make a decision
if z_score>critical_value:
    print("Reject null hypothesis")
else:
    print("Fail to reject null hypothesis")

Reject null hypothesis


###### The Z-score is 6.8745 This value is greater than the critical value of 1.64485, making the results statistically significant.

###### We can reject the null and conclude that the mean time working mothers spend talking to their children is does not equal 11. Based on the sample mean of 11.5, we know the mean time is higher.

##### Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. 
##### To test this claim, a sample of 40 customers is taken, and their wait times are recorded. 
##### The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes.
##### Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

In [5]:
import scipy.stats as stats
import numpy as np

In [6]:
population_mean=5   # population mean under null hypothesis
sample_size=40
sample_mean=4.6
population_std=0.8  # standard deviation of population
alpha=0.05           # significance level

In [7]:
# calculate t test statistics and critical value
t_score = (sample_mean-population_mean)/(population_std/np.sqrt(sample_size))
degree_freedom = sample_size-1
critical_value = stats.t.ppf(1-alpha,degree_freedom)
# print the z score and critical value
print("t_score : ",t_score)
print("Critical_value :",critical_value)

t_score :  -3.162277660168382
Critical_value : 1.6848751217112248


In [8]:
#CALCULATING P-VALUE
#We will use scipy stats function to calculate p value as shown below
p = stats.t.cdf(t_score, degree_freedom)
pvalue = stats.t.sf(np.abs(t_score), degree_freedom)*2
print("p is:", p)
print("pvalue is:", pvalue)


p is: 0.0015137112496894007
pvalue is: 0.0030274224993788014


In [9]:
# compare p value with alpha and make a decision
if pvalue > 0.05:
 print('Fail To Reject null hypothesis')
else:
 print('Reject null hypothesis ')

Reject null hypothesis 


###### So we reject Ho here & accept HA, our alternative hypothesis.

###### Therefore, we can easily say that there is enough evidence to support the coffee shop's claim that the average wait time for customers is less than 5 minutes.